File: backend/app/schemas/escalations.py
"""Schemas for EP-08 escalations (M8 Escalation System)."""
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import EscalationPriority, EscalationStatus


class EscalationCreate(BaseModel):
    conversation_id: uuid.UUID
    priority: EscalationPriority = EscalationPriority.medium
    reason: str = Field(..., max_length=500)
    escalation_type: str = Field(
        ...,
        max_length=50,
        description="voice_note | complex_issue | high_value_lead | payment_issue",
    )


class EscalationResponse(BaseModel):
    escalation_id: uuid.UUID
    conversation_id: uuid.UUID
    priority: EscalationPriority
    status: EscalationStatus
    reason: str
    escalation_type: str
    assigned_to: str | None = None
    created_at: datetime
    updated_at: datetime


class EscalationResolve(BaseModel):
    resolution_note: str = Field(..., max_length=1000)
    resolved_by: str = Field(..., max_length=100)


class EscalationResolveResponse(BaseModel):
    escalation_id: uuid.UUID
    status: EscalationStatus
    resolved_at: datetime

File: frontend/src/api/contracts/escalations.ts
export const operatorEscalationTypes = [
  'voice_note',
  'complex_issue',
  'high_value_lead',
  'payment_issue',
] as const

export type OperatorEscalationType = (typeof operatorEscalationTypes)[number]

export const operatorEscalationPriorities = ['low', 'medium', 'high'] as const

export type OperatorEscalationPriority =
  (typeof operatorEscalationPriorities)[number]

export interface OperatorEscalationCreate {
  reason: string
  type: OperatorEscalationType
  priority: OperatorEscalationPriority
}

export interface OperatorEscalationResponse extends OperatorEscalationCreate {
  escalation_id: string
  conversation_id: string
  status: 'open'
  source: 'operator_browser'
  created_at: string
  created_by: {
    account_id: string
    display_name: string
  }
}
